src/dft/backends/descriptor.cpp

```diff
@@ -28,7 +28,7 @@ namespace dft {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::commit(sycl::queue &queue) {
-    if (!pimpl_) {
+    if (!pimpl_ || pimpl_->get_queue() != queue) {
         pimpl_.reset(detail::create_commit(*this, queue));
```

Comment:
Next problem!

This frees the underlying commit. Doesn't that need to be kept alive for asynchronous compute calls that might still be in flight?

Reply:
That's true. You would hope any required underlying memory would be kept alive by the runtime, but I'm not sure there are actually guarantees for that.
What do you think about introducing a `pimpl_->get_queue().wait()`?

Reply:
I see two ways of doing this:
* Your suggested way - waiting on the queue (`pimpl_->get_queue().wait()`)
  * Blocks on changing descriptor queue.
* Shared pointers to the commit impl that get copied into the backend implementations
  * Non-blocking, but definitely not KISS.
  * Hard to know what should happen when querying commit status with this.

My preference would be for your suggestion.

Reply:
I have gone with the first. I don't really understand what would "hold" the shared_pointer other than the descriptor, and how would it know it was done? it would be a mess!